src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -250,22 +252,28 @@ private static FactoryContext CreateFactoryContext(RequestDelegateFactoryOptions
         return HandleRequestBodyAndCompileRequestDelegate(responseWritingMethodCall, factoryContext);
     }
 
-    private static RouteHandlerFilterDelegate CreateFilterPipeline(MethodInfo methodInfo, Expression? target, FactoryContext factoryContext)
+    private static RouteHandlerFilterDelegate CreateFilterPipeline(MethodInfo methodInfo, Expression? targetExpression, FactoryContext factoryContext, Expression<Func<HttpContext, object?>>? targetFactory)
     {
         Debug.Assert(factoryContext.Filters is not null);
         // httpContext.Response.StatusCode >= 400
         // ? Task.CompletedTask
-        // : handler((string)context.Parameters[0], (int)context.Parameters[1])
+        // : {
+        //  target = targetFactory(httpContext);
```

Comment:
this is confusing, where is target used?

Reply:
In `handler`, which might be something like `Convert(target).MethodName(parameters)`

Reply:
I added a note to the comment clarifying where the `target` comes in.